api/src/pcapi/core/offers/api.py

```diff
@@ -492,6 +492,12 @@ def batch_update_collective_offers_template(query: BaseQuery, update_fields: dic
         )
 
 
+def activate_future_offers(publication_date: datetime.datetime | None = None) -> None:
+    query = offers_repository.get_offers_by_publication_date(publication_date=publication_date)
```

Comment:
[détail / nommage] : ici je nommerais `publication_date` autrement, car on veut "publier" les offres dont la `publication_date`correspond à l'instant que l'on compare. Donc un truc genre `current_time` ou autre histoire de clarifier l'action effectuée

Reply:
et je sais pas à quel point c'est non-pythonic / moche, mais ça pourrait se faire de faire ça ?

```suggestion
def activate_future_offers(publication_date: datetime.datetime | None = datetime.datetime.utcnow() ) -> None:
    query = offers_repository.get_offers_by_publication_date(publication_date=publication_date)
```

Reply:
Sur le nommage `publication_date` ça me paraissait naturel de le nommer de cette façon. Il y a la fonction `get_offers_by_publication_date` avec un paramètre `publication_date` et du coup ce paramètre "se transmet" au fil des appels

Reply:
Pour le `datetime.datetime.utcnow()` en valeur par défaut, en plus du côté non-pythonic ça donnerait toujours la même datetime (évaluée au lancement de l'application), et on souhaite plutôt que la date de publication soit évaluée au moment où la fonction s'exécute :

```python
>>> import time
>>> from datetime import datetime
>>> 
>>> def func_1(dt: datetime = datetime.utcnow()):
...     print(dt)
... 
>>> 
>>> for _ in range(3):
...     func_1()
...     time.sleep(1)
... 
2024-06-26 15:17:29.989627
2024-06-26 15:17:29.989627
2024-06-26 15:17:29.989627
>>> 
>>> def func_2(dt: datetime | None = None):
...     if dt is None:
...         dt = datetime.utcnow()
...     print(dt)
... 
>>> 
>>> for _ in range(3):
...     func_2()
...     time.sleep(1)
... 
2024-06-26 15:18:52.595448
2024-06-26 15:18:53.600778
2024-06-26 15:18:54.604743
```